### PR TITLE
peers: support selfPeer for logical request

### PR DESCRIPTION
**Pull request by Raynos**
This change allows us to make a request to ourselves if
we put ourself in the peers list of a sub channel.

r: @shannili @jcorbin

### CODE REVIEW

**Comment by Raynos** [comment]
Python test failed ^

File: node/peers.js [hunk 1]
```diff
@@ -41,19 +41,24 @@ function TChannelPeers(channel, options) {
...
+        peers.push(self.selfPeer);
+    }
+    var counter = peers.length + 1;
     peers.forEach(function eachPeer(peer) {
         peer.close(onClose);
     });
     self.clear();
+    onClose();
 
     function onClose() {
```
Comment (ShanniLi) [inline node/peers.js]: why this function is necessary?

**Comment by ShanniLi** [comment]
lgtm.

File: node/peers.js (hunk 1 above)
Comment (Raynos) [inline node/peers.js]: I set counter to be length + 1 and we call `onClose` for each peer; I call an extra `onClose` at the end.

This means that Peers will close if there are zero peers;

The alternative is `if (peers.length === 0) { cb(); }`

**Pull request merged**